### ISSUE TITLE
Console error spam

### ISSUE THREAD

**Issue by SinsOfSolitude** [Zrips/CMI#5113]
So got this error spamming on repeat multiple times a second, unable to reproduce it. This has happened on half a dozen versions of Spigot, and about the same amount of latest Paper versions. CMI has been updated every time a new version comes out, and the errors still keep happening. Server is running on Ubuntu 16.04 LTS, tested on java version 11.0.7 and 15.0.2. The error seems to happen for one person at a time, they get timed out due to it. The error keeps endlessly spamming in the console on repeat multiple times a second, as I mentioned. Any time the person tries to reconnect, they will experience severe lag and inevitably time out again. Error spam persists no matter how much time has passed, largest log file was over 350MB. It happens to both players with good connectivity as well as bad. No plugins are interfering with it. I've tested this even on a completely clean server with no other plugins, completely vanilla settings. Weirdly, this isnt the case on my other server which hasnt had this happen even once, running on the same root server but earlier version of MC.

**ERROR (DELETE IF YOU HAVE NO ERROR):**
```
[16:46:50] [Netty Epoll Server IO #1/WARN]: java.nio.channels.ClosedChannelException
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:606)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.channel.DefaultChannelPipeline$HeadContext.close(DefaultChannelPipeline.java:1352)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.channel.AbstractChannelHandlerContext.invokeClose(AbstractChannelHandlerContext.java:622)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:606)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.handler.flush.FlushConsolidationHandler.close(FlushConsolidationHandler.java:173)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.channel.AbstractChannelHandlerContext.invokeClose(AbstractChannelHandlerContext.java:622)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:606)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:472)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.channel.DefaultChannelPipeline.close(DefaultChannelPipeline.java:957)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.channel.AbstractChannel.close(AbstractChannel.java:232)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at net.minecraft.server.v1_16_R3.NetworkManager.close(NetworkManager.java:431)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at net.minecraft.server.v1_16_R3.NetworkManager.exceptionCaught(NetworkManager.java:126)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:302)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:281)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.channel.AbstractChannelHandlerContext.fireExceptionCaught(AbstractChannelHandlerContext.java:273)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.handler.timeout.ReadTimeoutHandler.readTimedOut(ReadTimeoutHandler.java:98)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.handler.timeout.ReadTimeoutHandler.channelIdle(ReadTimeoutHandler.java:90)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.handler.timeout.IdleStateHandler$ReaderIdleTimeoutTask.run(IdleStateHandler.java:504)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.handler.timeout.IdleStateHandler$AbstractIdleTask.run(IdleStateHandler.java:476)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:170)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[16:46:50] [Netty Epoll Server IO #1/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:834)
```

---

**Cmi Version (using`/cmi version`):** 
Multiple versions, including the latest 8.8.1.2
**Server Type (Spigot/Paperspigot/etc):** 
Both Spigot and Paperspigot.
**Server Version (using `/ver`):** 
Paper 466

**Comment by Zrips** [comment]
I need full error log as this one doesn't even mention CMI in general so i cant really debug without any indication what and where it failed.

**Comment by SinsOfSolitude** [comment]
This is the only error I'm getting, and its this exact error on repeat. As I've mentioned I even tried running CMI alone across versions with the same result. Unsure what else I can provide :s

**Comment by Zrips** [comment]
There should be more of this one or something different at start. I cant really debug from message like this. have you tried testing it without any other plugin and only CMI?

**Comment by SinsOfSolitude** [comment]
Yeah I've done a completely vanilla setup with just CMI on it, fresh configs as well. Went through multiple logs and from start to finish, theres absolutely nothing except this :l

**Comment by Zrips** [comment]
Something seams to be different between those servers and as no one else reporting this issue it could be quite specific issue with your specific server setup. Hard to debug when nothing mentions CMI and it happens only on your single server without anyone else experiencing it. Maybe its issue with your server setup and CMI, maybe some specific player data causes issue, maybe players skin update is the issue. Hard to say at this point. If you could reproduce this issue consistently we could debug it properly.

**Comment by SinsOfSolitude** [comment]
I went back to CMI version 8.8.0.6 and Paper 444 on the server that's experiencing this issue and for the entirety of yesterday this problem didn't come up anymore. This is the same version combination I'm using for my main server which hasn't had this issue yet. I'll try to test each newer version of CMI and then the same for Paper and see where the issue comes up again.

**Comment by Zrips** [comment]
it would be nice to know which version causes this issue so we can investigate which change exactly caused it.

**Comment by SinsOfSolitude** [comment]
Okay so I've been testing, and it seems that some versions of Paper just have issues with CMI, and that might also be caused by my own system. This starts happening with Paper version 460 and ends at 472. Can't say for sure if it's definitely limited to this version gap, as it takes up to an hour or two for the error spam to come up sometimes. I also tried running the known faulty versions of Paper without CMI installed, and didn't experience the errors. From as far as I saw, the newer versions of Paper luckily don't seem to cause issues anymore, wasn't able to get them with Paper version 473 or newer. There's a chance this is partly caused by the drivers on my root server from what I was told by customer service at the data center, which I believe is indeed part of the problem, as I wasn't able to replicate this on 2 other machines. So this most likely should not happen to anyone else unless they have an identical system to mine.
